src/inmanta/data/__init__.py

```diff
@@ -5572,6 +5443,9 @@ async def update_persistent_state(
         connection: Optional[asyncpg.connection.Connection] = None,
         # TODO[#8541]: accept state.ResourceState and write blocked status as well
         last_deploy_result: Optional[state.DeployResult] = None,
+        blocked: Optional[str] = None,
+        is_undefined: Optional[bool] = None,
+        is_orphan: Optional[bool] = None,
```

Comment:
Does it make sense to also include these fields here? I added them because it was useful for a test case, but I could also do this in a separate method

Reply:
I'd prefer to leave them out. The scheduler purposely divides the table into intent-related columns and deploy-related ones, and is very careful about when to write to which. It might have been more explicitly documented, but that aside, I think we must be very careful to keep this separation in the helper methods as well.